### PR TITLE
use gradle lazy Provider for entrypoint container parameter

### LINKED ISSUE #4027: entrypoint cannot be set lazily on JibExtension in the Gradle plugin
While many properties on JibExtension utilise [lazy configuration APIs](https://docs.gradle.org/current/userguide/lazy_configuration.html), there are some, includingentrypoint, that do not. This forces users of the plugin to have to resort to using Project.afterEvaluate() in certain scenarios.

This issue is similar to #3927 but for entrypoint

### PR DESCRIPTION
**Pull request by wakingrufus**
use gradle lazy Provider for entrypoint container parameter (fixes #4027)

Thank you for your interest in contributing! For general guidelines, please refer to
the [contributing guide](https://github.com/GoogleContainerTools/jib/blob/master/CONTRIBUTING.md).

Before filing a pull request, make sure to do the following:

- [x] Create a new issue at https://github.com/GoogleContainerTools/jib/issues/new/choose.
- [x] Ensure that your implementation plan is approved by the team.
- [x] Verify that integration tests and unit tests are passing after the change.
- [x] Address all checkstyle issues. Refer to
  the [style guide](https://github.com/GoogleContainerTools/jib/blob/master/STYLE_GUIDE.md).

This helps to reduce the chance of having a pull request rejected.

### CODE REVIEW

**Comment by emmileaf** [comment]
@wakingrufus Generally, these workflows will be approved and run as part of reviewing the PR. We have a backlog of issues/feature requests and would appreciate your patience with some delay in reviewing.

That said, these presubmit checks run checkstyle, unit, and integration tests for the code changes, which can be verified locally. Please refer to the [contribution guide](https://github.com/GoogleContainerTools/jib/blob/master/CONTRIBUTING.md#typical-contribution-cycle) for more detailed instructions, if this is what you are looking to do for this PR. Thanks!

**Comment by wakingrufus** [comment]
I created https://github.com/GoogleContainerTools/jib/issues/4027 for this pr. I will rebase my PR to fix my commit message to properly link it

**Comment by wakingrufus** [comment]
@emmileaf Issue is created, linked, and PR is rebased.

File: jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ContainerParameters.java [hunk 1]
```diff
@@ -63,25 +63,29 @@ public ContainerParameters(ObjectFactory objectFactory) {
     creationTime = objectFactory.property(String.class).convention("EPOCH");
     mainClass = objectFactory.property(String.class);
     jvmFlags = objectFactory.listProperty(String.class);
+    entrypoint = objectFactory.listProperty(String.class);
   }
 
   @Input
-  @Nullable
```
Comment (emmileaf) [inline jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ContainerParameters.java:69]: Just a similar concern here for updating nullability for this public getter: https://github.com/GoogleContainerTools/jib/pull/3936#discussion_r1156115328

**Review by emmileaf — COMMENTED** [review]
The change look good overall, but can you add some test coverage for the lazy evaluation and a note on this enhancement in the README, similar to in https://github.com/GoogleContainerTools/jib/pull/3936? Thanks for the contribution, and apologies for the delay in reviewing!

File: jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ContainerParameters.java (hunk 1 above)
Comment (wakingrufus) [reply jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ContainerParameters.java]: Ok I updated this to keep in Nullable

**Comment by wakingrufus** [comment]
I updated the PR with changes to Docs and a Test.
Thanks again for taking a look at this!

**Review by emmileaf — COMMENTED** [review]
I don’t think the new `testLazyEvalForEntryPoint` unit test would run successfully - while the presubmit workflows need maintainer permissions to run, please make sure to verify your changes locally with `./gradlew build`.

Could you take a look into the `testProject` used in `JibPluginTest` and set up its [build.gradle](https://github.com/GoogleContainerTools/jib/blob/master/jib-gradle-plugin/src/test/resources/gradle/projects/lazy-evaluation/build.gradle#L36-L37) file to verify the feature added by this PR? (For example, in [testLazyEvalForLabels](https://github.com/GoogleContainerTools/jib/blob/1ecbced4bbfaaec0c9a580a34ad7b2202a7eba9f/jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/JibPluginTest.java#L364), the assertions verify that the [updated labels](https://github.com/GoogleContainerTools/jib/blob/1ecbced4bbfaaec0c9a580a34ad7b2202a7eba9f/jib-gradle-plugin/src/test/resources/gradle/projects/lazy-evaluation/build.gradle#L36-L37) specified in the test project’s `build.gradle` is lazy evaluated as expected (note the `project.ext.value` [change](https://github.com/GoogleContainerTools/jib/blob/1ecbced4bbfaaec0c9a580a34ad7b2202a7eba9f/jib-gradle-plugin/src/test/resources/gradle/projects/lazy-evaluation/build.gradle#L22) to "updated" in an `afterEvaluate` block.)  

Hope this helps!

File: jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ContainerParameters.java (hunk 1 above)
Comment (emmileaf) [reply jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ContainerParameters.java]: Thanks!

**Comment by wakingrufus** [comment]
> I don’t think the new `testLazyEvalForEntryPoint` unit test would run successfully - while the presubmit workflows need maintainer permissions to run, please make sure to verify your changes locally with `./gradlew build`.
> 
> Could you take a look into the `testProject` used in `JibPluginTest` and set up its [build.gradle](https://github.com/GoogleContainerTools/jib/blob/master/jib-gradle-plugin/src/test/resources/gradle/projects/lazy-evaluation/build.gradle) file to verify the feature added by this PR? (For example, in [testLazyEvalForLabels](https://github.com/GoogleContainerTools/jib/blob/1ecbced4bbfaaec0c9a580a34ad7b2202a7eba9f/jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/JibPluginTest.java#L364), the assertions verify that the [updated labels](https://github.com/GoogleContainerTools/jib/blob/1ecbced4bbfaaec0c9a580a34ad7b2202a7eba9f/jib-gradle-plugin/src/test/resources/gradle/projects/lazy-evaluation/build.gradle#L36-L37) specified in the test project’s `build.gradle` is lazy evaluated as expected (note the `project.ext.value` [change](https://github.com/GoogleContainerTools/jib/blob/1ecbced4bbfaaec0c9a580a34ad7b2202a7eba9f/jib-gradle-plugin/src/test/resources/gradle/projects/lazy-evaluation/build.gradle#L22) to "updated" in an `afterEvaluate` block.)
> 
> Hope this helps!

Sorry about that. Thanks for the guidance. I have updated the test and verified it is passing locally.
Thanks again @emmileaf !

**Review by emmileaf — APPROVED** [review]
LGTM - thanks for the contribution @wakingrufus!

**Pull request merged**